src/subcommand/fix.rs

```diff
@@ -270,16 +270,33 @@ impl Error {
     }
 }
 
+/// A logger. This is used to print the output to stdout
+pub trait Logger {
+    /// Log things. This is mutable because TestLogger mutates when we log
+    fn log(&mut self, content: &str);
+}
```

Comment:
Generally speaking, we should do our best to not impose a worse API on the non-test code path due to testing concerns. In particular, the issue with this method being `mut` is that a single logger cannot be provided to multiple users at the same time; this is a clear antipattern for the purposes in which someone would use a logging provider.

In tests, we can use the [interior mutability pattern](https://doc.rust-lang.org/book/ch15-05-interior-mutability.html) to support a non-`mut` `log` method:
```rust
/// A logger that prints to stdout and also keeps track of what has been logged so that we can test it.
#[derive(Default)]
struct TestLogger {
    output: RwLock<Vec<String>>,
}

impl TestLogger {
    fn output(&self) -> String {
        self.output
            .read()
            .expect("lock must not be poisoned")
            .join("\n")
    }

    fn new() -> Self {
        Default::default()
    }
}

impl Logger for TestLogger {
    fn log(&self, content: &str) {
        let mut guard = self.output.write().expect("lock must not be poisoned");
        guard.deref_mut().push(content.to_string());
    }
}
```
(The link references `RefCell`, which is a single-threaded way to use this pattern; here we use `RwLock` to make our implementation thread safe).

Please redesign the trait and impls with this in mind.